### PR TITLE
Update the command to use IPython notebook

### PR DESCRIPTION
**Pull request by yuecong**
As for "notebook --pylab inline" is not supported any more, update the related documentation for this.

### CODE REVIEW

File: docs/programming-guide.md [hunk 1]
```diff
@@ -237,9 +237,11 @@ You can customize the `ipython` command by setting `PYSPARK_DRIVER_PYTHON_OPTS`.
 the [IPython Notebook](http://ipython.org/notebook.html) with PyLab plot support:
 
 {% highlight bash %}
-$ PYSPARK_DRIVER_PYTHON=ipython PYSPARK_DRIVER_PYTHON_OPTS="notebook --pylab inline" ./bin/pyspark
+$ PYSPARK_DRIVER_PYTHON=ipython PYSPARK_DRIVER_PYTHON_OPTS="notebook" ./bin/pyspark
 {% endhighlight %}
 
+After IPython Notebook server is launched, you can create a new "Python 2" notebook from "Files" tab. Inside the notebook, you can input the '%pylab inline' command as part of your notebook before you start to try spark from IPython notebook.
```
Comment (srowen) [inline docs/programming-guide.md]: Just out of curiosity, it's not supported as of what version -- something old, or recent?
Nit: you can back-tick-quote the `%pylab inline` bit.
Is it worth explaining what this option does, then? (Could be entirely obvious, just asking.)

**Comment by yuecong** [comment]
It is the IPython 3.0. I believe one of the motivation of IPython notebook is their chart drawing functionality. This is the reason why pylab library is specially mentioned.

**Comment by srowen** [comment]
But does this then work with ipython 2? I wouldn't want to necessarily 'break' support, even if it's just in an example. Or are two examples called for? Ideally, one example is good, even if it's deprecated in new ipython versions.

**Comment by yuecong** [comment]
The notebook UI is different between 2.4 and new 3.0. But %pylab inline both work for 2.4 and 3.0 version. 
I agree that we do not need to explain how to launch a notebook UI from IPython. Instead, just tell the users they can add `%pylab inline` into the notobook to use pylab library.
Do I need to update the patch then or the admin will handle with this?

**Comment by srowen** [comment]
OK, if this isn't making the example unusable for reasonably recent versions of ipython then it seems OK. Oh, above I was not suggesting that you remove your additional documentation but instead that you explain what the "inline" option was all about? or will it be obvious already?

**Comment by yuecong** [comment]
Let me clarify my opinions more clearly.
1, change '$ PYSPARK_DRIVER_PYTHON=ipython PYSPARK_DRIVER_PYTHON_OPTS="notebook --pylab inline" ./bin/pyspark' to $ PYSPARK_DRIVER_PYTHON=ipython PYSPARK_DRIVER_PYTHON_OPTS="notebook " ./bin/pyspark'.
 For this, we agree it as for it will not work any more for ipython 3.0
2, Whether it is necessary to methion '%pylab inline' or not. 
  I think it is necessary, as for this give the users to understand that with ipython notebook, they can visualize their data with pylab, which is different from ipython shell.
3, Whether it needs to add how to launch a notebook from ipython notebook UI.
 Originally, I add the explanation on the base of ipython3.0, but as you commented, I find the ipython notebook UI is different between 2.x and 3.x, so I agree we may do not need to explain it in detail to make the guide be suitable for all version of ipythons.

Hope the above can clarify my opinions. :)

**Comment by srowen** [comment]
OK, nevermind my question. I think it's clear you know what to do here and it's as you think it should be. I'll leave it open a bit for any other opinions but if it's making the example work for more ipython versions, fine.

**Comment by srowen** [comment]
OK I merged this with a couple fixes to the text (e.g. spark -> Spark)

**Comment by yuecong** [comment]
Thanks.

**Pull request closed without merging**